zallet/src/commands/migrate_zcash_conf.rs

```diff
@@ -0,0 +1,644 @@
+            Action::warn(|_| {
+                Some(fl!(
+                    "migrate-warn-cli-only",
+                    option = "zapwallettxes",
+                    flag = "--zap-txes=MODE",
+                ))
+            }),
+        )));
+
+    // Documented wallet debugging/testing options.
+    let documented_wallet_debug_options = iter::empty()
+        // `dblogsize` doesn't have an SQLite analogue.
+        .chain(Action::ignore("dblogsize"))
+        // `flushwallet` does have SQLite analogues, but they have different semantics, so
+        // we aren't migrating this across by default.
+        .chain(Some((
+            "flushwallet",
+            Action::warn(|value| {
+                (value != "1").then(|| {
+                    fl!(
+                        "migrate-warn-unsupported",
+                        option = "flushwallet",
+                        value = value,
+                    )
+                })
+            }),
+        )))
+        // TODO: Figure out if SQLite has an analogue for BDB's `DB_PRIVATE`.
+        .chain(Action::ignore("privdb"));
+
+    // Undocumented options used only by wallet code.
+    let undocumented_wallet_options = iter::empty()
+        // TODO: Decide if we want to map this to the eventual "untrusted confirmations"
+        // setting for the improved note selection logic.
+        .chain(Action::ignore("anchorconfirmations"))
+        // Unsupported in `zcashd` since 5.5.0.
+        .chain(Action::ignore("mintxfee"))
+        // TODO: Determine whether we need this for regtest testing of Zallet.
+        .chain(Action::ignore("regtestwalletsetbestchaineveryblock"))
+        // Unsupported in `zcashd` since 5.5.0.
+        .chain(Action::ignore("sendfreetransactions"))
+        // Unsupported in `zcashd` since 5.5.0.
+        .chain(Action::ignore("txconfirmtarget"));
+
+    // Node options used directly by wallet code.
+    let node_options_direct_wallet = iter::empty()
+        // Experimental feature we aren't migrating.
+        .chain(Action::ignore("developerencryptwallet"))
+        // Used to re-enable CPU mining after disabling it during proving.
+        // Irrelevant for Zallet which doesn't include mining.
+        .chain(Action::ignore("genproclimit"))
+        // TODO: Figure out where this was used, and if we want to keep it.
+        .chain(Action::ignore("maxtxfee"))
+        // Used to check whether the configured miner address exists in the wallet.
+        // Irrelevant for Zallet which doesn't include mining.
+        .chain(Action::ignore("mineraddress"))
+        // Experimental feature we aren't migrating.
+        .chain(Action::ignore("paymentdisclosure"))
+        .chain(Some((
+            "preferredtxversion",
+            Action::warn(|value| {
+                Some(fl!(
+                    "migrate-warn-unsupported",
+                    option = "preferredtxversion",
+                    value = value,
+                ))
+            }),
```

Comment:
There is no need to warn if the preferred version is 5, since that's the default for zallet (and will continue to be the default immediately after NU7 I think?)